raft/raft.go

```diff
@@ -1566,8 +1566,8 @@ func (r *raft) switchToConfig(cfg tracker.Config, prs tracker.ProgressMap) pb.Co
 			r.maybeSendAppend(id, false /* sendIfEmpty */)
 		})
 	}
-	// If the the leadTransferee was removed, abort the leadership transfer.
-	if _, tOK := r.prs.Progress[r.leadTransferee]; !tOK && r.leadTransferee != 0 {
+	// If the the leadTransferee was removed or demoted, abort the leadership transfer.
```

Comment:
can you add a test?

Reply:
Yes, but I'm not sure the test case should be in raft_test.go or testdata/*.txt.

Reply:
i think there should be a related test case for the removal one. adding a case there should be fine.

Reply:
Thank you! I will update the PR soon.